doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -984,10 +984,12 @@ Helper tactics
 
 .. tacn:: decide equality
 
-   Solves a goal of the form :g:`forall x y : R, {x = y} + {~ x = y}`,
-   where :g:`R` is an inductive type such that its constructors do not take
-   proofs or functions as arguments, nor objects in dependent types. It
-   solves goals of the form :g:`{x = y} + {~ x = y}` as well.
+   Solves a goal of the form :g:`forall x y : R, {x = y} + {~ x = y}` or
+   :g:`forall x y : R, (x = y) \/ (~ x = y)`, where :g:`R` is an
+   inductive type such that its constructors do not take proofs
+   or functions as arguments, nor objects in dependent types. It
```

Comment:
```suggestion
   inductive type whose constructors do not take proofs
   or functions as arguments, nor objects in dependent types. It
```

Optiona: Is there better wording for "proofs or functions as arguments, nor objects in dependent types"?  It's a bit opaque to me.  I don't recall other references to proofs as arguments in the doc--proofs are just terms.  It's also better to state what is accepted first before saying what's not accepted.  "nor objects in dependent types" - in arguments or not in arguments?  (Note I've not used and don't fully understand the tactic.  Or dependent types.)  @Alizter perhaps you can suggest better wording?

Reply:
@Alizter FWIW, probably better if you had made your last comment [below](https://github.com/coq/coq/pull/16461#issuecomment-1244092632) as a reply to the review comment, closer to what it responds to.

`(B a) -> I` is functions, makes sense
`A : Prop |- A -> I` looks like a proof but I don't see `|-` in the grammar for terms nor do I see it defined as a notation.
Then `(a -> b) -> I` must be "objects in dependent types", which I don't understand.  What is an "object"?  That's not common terminology.  Also, how do you know if it's dependent or non-dependent?  Perhaps that depends on what a and b are.

@Zimmi48 Perhaps you would suggest some wording?

Or we can just leave it as it is for now.

Reply:
@jfehrle Sorry `|-` is just short hand for the context. So I meant something like `forall (A : Prop), A -> I` or `A -> I` where `A : Prop` in the context.

Reply:
`B a` is "objects in dependent types". (`B` is dependent on the value of `a`)
`(a -> b)` is functions. (the type of functions `a -> b`).